src/screen.h

```diff
@@ -229,20 +231,48 @@ class cached_esc_sequences_t {
 #endif
 
         cache.emplace(str);
-        lengths.insert(len);
+        if (std::find(lengths.begin(), lengths.end(), len) == lengths.end()) {
+            lengths.push_back(len);
+            lengths_match_count[len] = 0;
+        }
     }
 
+    // This is used to sort the cached sequence lengths from most frequently matched to least.
+    class custom_cmp {
+       private:
+        std::map<size_t, size_t> &lengths_match_count;
+
+       public:
+        explicit custom_cmp(std::map<size_t, size_t> &x) : lengths_match_count(x){};
+
+        bool operator()(size_t l1, size_t l2) const {
+            return lengths_match_count[l1] > lengths_match_count[l2];
+        }
+    };
+
     size_t find_entry(const wchar_t *entry) {
         for (auto it : lengths) {
             auto match = cache.find(wcstring(entry, it));
-            if (match != cache.end()) return it;
+            if (match != cache.end()) {  // we found a matching cached sequence
+                // Periodically sort  the sequence lengths so we check for matches going from the
+                // most frequently matching lengths to least frequent.
+                lengths_match_count[it]++;
+                if (++cache_hits % 1000 == 0) {
+                    std::sort(lengths.begin(), lengths.end(), custom_cmp(lengths_match_count));
```

Comment:
Thx. I had tried that but got the following error because I forgot the ampersand to capture local names by reference: `'this' cannot be implicitly captured in this context`.